ligo-webide-backend/src/Lib.hs

```diff
@@ -34,17 +34,53 @@ import System.IO.Temp
 import System.Process
 import System.Exit (ExitCode (ExitSuccess, ExitFailure))
 
-type API = "compile" :> ReqBody '[JSON] Value :> Post '[JSON] Text
+-- Swagger
+import Data.Proxy (Proxy (..))
+import Servant.Swagger
+import Data.Swagger.Schema (toSchema, ToSchema)
+import Data.Swagger.ParamSchema (ToParamSchema)
+import Servant.Swagger.UI
+
+type API = "compile" :> ReqBody '[JSON] CompileRequest :> Post '[JSON] Build
+type SwaggeredAPI = SwaggerSchemaUI "swagger-ui" "swagger.json"
+                  :<|> API
+
+newtype Build = Build Text deriving stock (Show, Generic)
+
+instance ToJSON Build where
+  toJSON = genericToJSON defaultOptions
+
+instance ToSchema CompileRequest
+instance ToSchema Build
+instance ToParamSchema Build
 
 data Config = Config
   { cLigoPath :: FilePath
   , cPort :: Int
   , cVerbose :: Bool
   }
 
+newtype Source = Source {unSource :: Text} deriving stock (Eq, Show, Ord, Generic)
+
+instance ToSchema Source
+instance ToSchema File
+
+instance FromJSON Source where
+  parseJSON = genericParseJSON defaultOptions
+instance ToJSON Source where
+  toJSON = genericToJSON defaultOptions
+
+newtype File = File Text deriving stock (Eq, Show, Ord, Generic)
+
+instance FromJSON File where
+  parseJSON = genericParseJSON defaultOptions
+instance ToJSON File where
+  toJSON = genericToJSON defaultOptions
+
 data CompileRequest = CompileRequest
   { rFileExtension :: Text
-  , rSource :: Text
+  , rSources :: [(File, Source)]
+  , rMain :: File
   } deriving stock (Eq, Show, Ord, Generic)
```

Comment:
@sashasashasasha151 Do you think this simple POST body is a good type for `CompileRequest`? Or should I think about MIME/multipart requests?

Reply:
Okay, maybe let's do not think about `MIME/multipart` when all works fine \
As it future we are planning to not use backend, maybe it will be overengineering right now? \
Also as I understand, yes, it is possible to store files as files, but right now all code is simply using strings \
So, for only thins request I think we can use just POST, but maybe later we will reimplement it

Reply:
Yup, sounds good. I'll also close this PR and move it to GitLab since we should do new development there